src/FinfoMimeTypeDetector.php

```diff
@@ -39,23 +39,38 @@ class FinfoMimeTypeDetector implements MimeTypeDetector, ExtensionLookup
      */
     private $inconclusiveMimetypes;
 
+    /**
+     * @var bool
+     */
+    private $ignoreNonSeekableStreams;
+
+    /**
+     * Buffer size to read from streams if no other bufferSampleSize is defined.
+     */
+    const STREAM_BUFFER_SAMPLE_SIZE_DEFAULT = 4100;
+
     public function __construct(
         string $magicFile = '',
         ExtensionToMimeTypeMap $extensionMap = null,
         ?int $bufferSampleSize = null,
-        array $inconclusiveMimetypes = self::INCONCLUSIVE_MIME_TYPES
+        array $inconclusiveMimetypes = self::INCONCLUSIVE_MIME_TYPES,
+        bool $ignoreNonSeekableStreams = true
     ) {
         $this->finfo = new finfo(FILEINFO_MIME_TYPE, $magicFile);
         $this->extensionMap = $extensionMap ?: new GeneratedExtensionToMimeTypeMap();
         $this->bufferSampleSize = $bufferSampleSize;
         $this->inconclusiveMimetypes = $inconclusiveMimetypes;
+        $this->ignoreNonSeekableStreams = $ignoreNonSeekableStreams;
     }
 
     public function detectMimeType(string $path, $contents): ?string
     {
-        $mimeType = is_string($contents)
-            ? (@$this->finfo->buffer($this->takeSample($contents)) ?: null)
-            : null;
+        $mimeType = null;
+        if (is_string($contents)) {
+            $mimeType = @$this->finfo->buffer($this->takeSample($contents));
+        } elseif (is_resource($contents)) {
```

Comment:
Instead of detecting if the stream is seekable in `takeSample` let's extract it and us it in the if condition. Basically, I'd want the sampling only to happen when the stream is seekable, otherwise get the current behaviour, otherwise it's a BC break and I don't think this warrants it.

Reply:
Done. However, not sure if the notation should be shortened to `!empty(stream_get_meta_data($contents)['seekable'])` - am not sure about php behavior in all supported versions nor do I know your personal preference :).